client/components/sites-dropdown/index.jsx

```diff
@@ -39,27 +42,23 @@ export class SitesDropdown extends PureComponent {
 	constructor( props ) {
 		super( props );
 
+		// needed to be done in constructor b/c spy tests
 		this.selectSite = this.selectSite.bind( this );
 		this.toggleOpen = this.toggleOpen.bind( this );
 		this.onClose = this.onClose.bind( this );
-
-		const selectedSite = props.selectedSiteId
-			? props.getSite( props.selectedSiteId )
-			: props.getSite( props.primarySiteId );
-
-		this.state = {
-			selectedSiteSlug: selectedSite && selectedSite.slug
-		};
 	}
 
-	getSelectedSite() {
-		return this.props.getSite( this.state.selectedSiteSlug );
+	state = { open: false }
+
+	componentDidMount() {
+		const { initialSiteId, setSelectedSiteSlug } = this.props;
+		setSelectedSiteSlug( initialSiteId );
```

Comment:
Actually, ID on its own would clash with a prop that is part of `SitesDropdown`'s public interface, so I'm changing to `locallySelectedSiteId`.